include/common/core/spi.hpp

```diff
@@ -26,8 +26,8 @@ class TMC2130Spi {
      * @param transmit The transmit buffer.
      * @param receive The receive buffer.
      */
-    virtual void transmit_receive(const TMC2130Spi::BufferType& transmit,
-                                  TMC2130Spi::BufferType& receive) = 0;
+    virtual auto transmit_receive(const TMC2130Spi::BufferType& transmit,
```

Comment:
As we are already refactoring in this PR, I think it's time we call this thing something other than `TMC2130Spi` cus it's getting a bit confusing. We could instead call it `SpiDeviceBase` similar to `I2CDeviceBase`. For now, let's keep the `build_command` as is. If we find we need to change it, we can.

Reply:
I think `build_command` definitely needs to be updated in the future, but I don't care to do it in this PR. A good time to work on that might be when we refactor spi to make it its own task.

Reply:
Yep I agree. I think it would just be nice to at least change the name of this base class in this PR b/c we now have a class called TMC2130 and it's a bit confusing.

Reply:
I think that's a good idea - no code changes other than making it `class SpiBase`.